packages/x-data-grid/src/components/virtualization/GridVirtualScrollerRenderZone.tsx

```diff
@@ -37,14 +34,10 @@ const GridVirtualScrollerRenderZone = forwardRef<
   React.HTMLAttributes<HTMLDivElement> & { sx?: SxProps<Theme> }
 >(function GridVirtualScrollerRenderZone(props, ref) {
   const { className, ...other } = props;
-  const apiRef = useGridApiContext();
+  const apiRef = useGridPrivateApiContext();
   const rootProps = useGridRootProps();
   const classes = useUtilityClasses(rootProps);
-  const offsetTop = useGridSelector(apiRef, () => {
-    const renderContext = gridRenderContextSelector(apiRef);
-    const rowsMeta = gridRowsMetaSelector(apiRef);
-    return rowsMeta.positions[renderContext.firstRowIndex] ?? 0;
-  });
+  const offsetTop = apiRef.current.virtualizer.api.useVirtualization().getters.getOffsetTop();
```

Comment:
Why not do this instead? Seems more clear to me:

```suggestion
  const renderContext = useStore(
    apiRef.current.virtualizer.store,
    Virtualization.selectors.renderContext,
  );
  const rowsMeta = useStore(apiRef.current.virtualizer.store, Dimensions.selectors.rowsMeta);
  const offsetTop = rowsMeta.positions[renderContext.firstRowIndex] ?? 0;
```

Reply:
Answered above.